executor/analyze.go

```diff
@@ -279,7 +283,7 @@ func (e *AnalyzeExec) handleResultsErrorWithConcurrency(ctx context.Context, sta
 	saveResultsCh := make(chan *statistics.AnalyzeResults, partitionStatsConcurrency)
 	errCh := make(chan error, partitionStatsConcurrency)
 	for i := 0; i < partitionStatsConcurrency; i++ {
-		worker := newAnalyzeSaveStatsWorker(saveResultsCh, subSctxs[i], errCh)
+		worker := newAnalyzeSaveStatsWorker(saveResultsCh, subSctxs[i], errCh, &e.ctx.GetSessionVars().Killed)
```

Comment:
Do we need to check killed in `for panicCnt < statsConcurrency` loop? If all the workers sense the killed signal and exit, no one consumes `saveResultsCh` and when it is full, we would be stuck at `saveResultsCh <- results`.

Reply:
updated.